server/gokbg3/grails-app/init/gokbg3/BootStrap.groovy

```diff
@@ -995,6 +999,7 @@ class BootStrap {
         RefdataCategory.lookupOrCreate('Job.Type', 'PlatformCleanup').save(flush: true, failOnError: true)
         RefdataCategory.lookupOrCreate('Job.Type', 'RecalculateStatistics').save(flush: true, failOnError: true)
 
+        LanguagesService.initialize()
```

Comment:
Calling the LanguagesService here without setting a default URL for it in `application.yml` seems bad. Either have it not initialize at all by default and ensure some fallback values, or provide a working default.

Reply:
@hornmo This was implemented likewise the `ygorUrl` example in application.yml. Do you think `ygorUrl` should be changed to a default URL as well? I thought this `null` implementation was done by intention because in reality, there is no default value for a server-specific URL. Hard-coded fallback URLs seem to be the least good solution since they are always a source of misconfiguration when changing the setup.

Reply:
As it stands, if no ygorUrl is provided the functionality is just not enabled at all. This on the other hand will throw an exception and stop the application from starting at all. What I meant as a fallback would be to define a small sample of RefdataValues (some commonly used languages) to be created instead if no URL is defined.

Reply:
This holds for the `languagesUrl`. If it is misconfigured and pointing towards a non-existing address, `LanguagesService.initialize()` will throw an exception when being called in `Bootstrap.refdataCats()`. The same holds for its value being `null`. GOKb will simply not start then.

Reply:
Throwing an exception for an erroneous config value sounds sensible. Using the default application config though should not prevent the application from starting (especially relevant for integration testing).

Reply:
Like we've discussed, I've changed the config value to `localhost:8070` and added the necessary documentation in the [readme](1becddb89394ef27288c51739b81f7ba51476722) and [the wiki site](https://github.com/openlibraryenvironment/gokb/wiki/GOKb-8.x-Installation-and-Bootstrapping-Guide).